packages/proxy/test/unit/http/response-middleware.spec.ts

```diff
@@ -2098,4 +2098,71 @@ describe('http/response-middleware', function () {
       }
     }
   })
+
+  describe('GzipBody', function () {
+    const { GzipBody } = ResponseMiddleware
+    let ctx
+    let responseStreamReceivedStub
+    let makeResStreamPlainTextStub
+
+    beforeEach(() => {
+      responseStreamReceivedStub = sinon.stub()
+      makeResStreamPlainTextStub = sinon.stub()
+    })
+
+    it('calls responseStreamReceived on protocolManager if protocolManager present and request is correlated', function () {
```

Comment:
Suggest adding a test for when the request is not correlated and when the protocolManager hasn't been set.

Reply:
Added: [`580485e` (#27462)](https://github.com/cypress-io/cypress/pull/27462/commits/580485e928fa4e3588e6720359dcb38f7a0aaa92)